excel.py

```diff
@@ -0,0 +1,56 @@
+#! python3
+# Reads an excel file and transforms to JSON.
+
+import xlrd
+from collections import OrderedDict
+import simplejson as json
+import psycopg2
+import urllib.parse
+
+# set up the connection. Extract the database info from the URI and connect with psycopg2
+def connect():
+    try:
+        result = urllib.parse.urlparse("postgres://4cebab44:4j602JWnaP3Yaw@db.nuodata.io:5432/4cebab44")
+        username = result.username
+        password = result.password
+        database = result.path[1:]
+        hostname = result.hostname
+        DB = psycopg2.connect(
+            database = database,
+            user = username,
+            password = password,
+            host = hostname
+        )
+        c = DB.cursor()
+        return DB, c
+    except:
+        print("<couldn't connect>")
```

Comment:
You've just created a function connect() that silently fails if a failure happens.

Best practice is to capture and log the full error information and rethrow a custom exception to force the caller to handle it.